js.go

```diff
@@ -2448,11 +2456,18 @@ func checkMsg(msg *Msg, checkSts bool) (usrMsg bool, err error) {
 		// 404 indicates that there are no messages.
 		err = errNoMessages
 	case reqTimeoutSts:
-		// Older servers may send a 408 when a request in the server was expired
-		// and interest is still found, which will be the case for our
-		// implementation. Regardless, ignore 408 errors until receiving at least
-		// one message.
-		err = ErrTimeout
+		// In case of a fetch request with no wait request and expires time,
+		// it will be a 408 error but with a Requests Pending description.
+		desc := msg.Header.Get(descrHdr)
```

Comment:
From looking at server code, there are several possible descriptions for 408 (so far..): "Interest Expired", "Requests Pending", "Request Canceled" and "Request Timeout".
We need to think if we need to behave any differently for those error status or not. I am not sure. Depending on that answer, we can decide if we need to distinguish description or not.. (again, there are more than that one, and some may be added in the future).

Reply:
Not sure about the rest of the status at the moment but I think one way to simplify this could be to refactor the code to skip 408s unless making the initial expires + no wait batch request.

Reply:
All of those mean the request is no longer valid and no additional messages will be sent for that request.